File: LeetCode/Java Solutions/Delete Nodes From Linked List Present in Array/delete.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        // Create a set of numbers to remove
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }
        
        // Create a dummy node to handle the case where the head needs to be removed
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode current = dummy;
        
        // Traverse the list and remove nodes with values in numSet
        while (current.next != null) {
            if (numSet.contains(current.next.val)) {
                current.next = current.next.next;
            } else {
                current = current.next;
            }
        }
        
        return dummy.next;
    }
}

File: LeetCode/Python Solutions/Delete Nodes From Linked List Present in Array/delete.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        # Create a set of numbers to remove
        num_set = set(nums)
        
        # Create a dummy node to handle the case where the head needs to be removed
        dummy = ListNode(0)
        dummy.next = head
        current = dummy
        
        # Traverse the list and remove nodes with values in num_set
        while current.next:
            if current.next.val in num_set:
                current.next = current.next.next
            else:
                current = current.next
        
        return dummy.next